File: Arrays/EncodeDecode/encodeDecode.py
def encode(strings):
    res = []
    for string in strings:
        res.append(str(len(string)))
        res.append("#")
        res.append(string)
    return "".join(res)

def decode(string):
    res, left, i = [], 0, 0
    
    while i < len(string):
        c = string[i]
        if c != "#":
            i += 1
            continue
        length = int(string[left:i])
        res.append(string[i+1:i+length+1])
        i += 1 + length
        left = i
    return res

test = ["lint","code","love","you"]
res = encode(test)
print(res)
res2 = decode(res)
print(res2)

File: Arrays/EncodeDecode/encodeDecode.go
package main

import (
	"bytes"
	"fmt"
	"strconv"
)

func encode(words []string) string {
	var buffer bytes.Buffer
	for _, word := range words {
		buffer.WriteString(fmt.Sprintf("%v", (word)))
		buffer.WriteString("#")
		buffer.WriteString(word)
	}
	return buffer.String()
}

func decode(word string) []string {
	res := []string{}
	i, left := 0, 0
	for i < len(word) {
		c := word[i]
		if c != '#' {
			i += 1
			continue
		}
		length, err := strconv.Atoi(word[left:i])
		if err != nil {
			print(err)
		}

		currWord := word[i+1 : i+length+1]
		res = append(res, currWord)
	}
	return res
}
